pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/flags.go

```diff
@@ -230,8 +231,12 @@ func admissionFlags(admissionPluginConfig map[string]*configapi.AdmissionPluginC
 
 			continue
 		}
+
 		// if it wasn't a DefaultAdmissionConfig object, let the plugin deal with it
-		if defaultConfig, ok := config.Configuration.(*configapi.DefaultAdmissionConfig); !ok {
+		currConfig := &configapi.DefaultAdmissionConfig{}
+		uncastDefaultConfig, _, decodingErr := configapilatest.Codec.Decode(config.Configuration.Raw, nil, currConfig)
+		defaultConfig, ok := uncastDefaultConfig.(*configapi.DefaultAdmissionConfig)
```

Comment:
won't this panic on nil `uncastDefaultConfig`?

Reply:
> won't this panic on nil uncastDefaultConfig?

I'll defend and see if there's a good way to write a test for it.

Reply:
just check err first, cast later.